src/realtime-callbacks.js

```diff
@@ -56,6 +56,91 @@ module.exports.setNow = function(f) {
 };
 let _now = Date.now;
 
+// Internal hash of active debouncers
+const _debouncers = {};
+
+class Debouncer {
+
+    constructor(jobName) {
+        this.jobName = jobName;
+    }
+
+    complete() {
+        delete _debouncers[this.jobName];
+        this.func();
+    }
+
+    start(func, waitMs) {
+        if (func) this.func = func;
+        if (!isNaN(waitMs)) this.waitMs = waitMs;
+        this.timeoutId = exports.setTimeout(this.complete.bind(this), this.waitMs);
+    }
+
+    stop() {
+        if (this.timeoutId !== null || this.timeoutId !== undefined) {
+            exports.clearTimeout(this.timeoutId);
+            delete this.timeoutId;
+        }
+    }
+}
+
+/**
+ * Implementation of debounce a which will call the callback after
+ * a waitMs period if it is not debounced within that period.
+ *
+ * After a jobName has been defined it may be called without
+ * {func} or {waitMs} arguments to debounce with initial settings.
+ * New {func} or {waitMs} params will over-write ones previously
+ * set in the debouncer.
+ * If called without a jobName a random one is generated.
+ *
+ * @param {String} jobName  a jobName to reference the debouncer
+ * @param {function} func   callback to be called after a delay
+ * @param {Number} waitMs   number of milliseconds to delay by
+ *
+ * @return {String} jobName a jobName to reference the debouncer
+ */
+
+module.exports.debounce = function(jobName, func, waitMs) {
+    if (jobName === undefined || jobName === null) {
```

Comment:
I guess jobName could be `0`? If there is no reasonable case for that than `!` should be fine.